sway-book/src/documentation/language/variables/const.md

```diff
@@ -0,0 +1,17 @@
+# const
+
+A `const` is similar to an [immutable let](./let.md#immutable) however there are a few differences.
+
+- The constant is evaluated at compile-time
+- A constant can be declared in any scope i.e. inside of a [function](../functions/index.md) and outside
+- The `mut` keyword cannot be used
+
+## Declaration
+
+To define a constant the `const` keyword is used followed by a name and an assignment of a value.
```

Comment:
```suggestion
To define a constant, the `const` keyword is used followed by a name and an assignment of a value. Providing the type of a `const` is optional.
```

Reply:
What's the purpose of mentioning the `optional` constant?